### PR TITLE
Refactor ColumnarReadNextRow for better readability

### PR DESCRIPTION
**Pull request by onurctirtir**

### CODE REVIEW

File: src/backend/columnar/columnar_reader.c [hunk 1]
```diff
@@ -226,6 +220,28 @@ ColumnarReadNextRow(ColumnarReadState *readState, Datum *columnValues, bool *col
...
+ */
+static bool
+StripeReadInProgress(ColumnarReadState *readState)
+{
+	return readState->stripeReadState != NULL;
+}
+
+
+/*
+ * HasUnreadStripe return true if we still have stripes to read during current
```
Comment (SaitTalhaNisanci) [inline src/backend/columnar/columnar_reader.c:234]: returns

File: src/backend/columnar/columnar_reader.c [hunk 2]
```diff
@@ -288,12 +304,19 @@ BeginStripeRead(StripeMetadata *stripeMetadata, Relation rel, TupleDesc tupleDes
...
 {
-	pfree(stripeReadState);
+	readState->chunkGroupsFiltered +=
+		readState->stripeReadState->chunkGroupsFiltered;
+	pfree(readState->stripeReadState);
+	readState->stripeReadState = NULL;
+	MemoryContextReset(readState->stripeReadContext);
+
+	readState->currentStripe++;
 }
```
Comment (jeff-davis) [inline src/backend/columnar/columnar_reader.c:320]: I'd prefer to keep destruction of the stripeReadState separate from advancing to a new stripe. We could keep EndStripeRead as-is, and create a new function AdvanceStripe or something?

File: src/backend/columnar/columnar_reader.c (hunk 1 above)
Comment (onurctirtir) [reply src/backend/columnar/columnar_reader.c]: d736745

File: src/backend/columnar/columnar_reader.c (hunk 2 above)
Comment (onurctirtir) [reply src/backend/columnar/columnar_reader.c]: makes sense, what about b007d242300501afee9dcc9a572901ad5154450b ?

File: src/backend/columnar/columnar_reader.c [hunk 3]
```diff
@@ -283,12 +301,27 @@ BeginStripeRead(StripeMetadata *stripeMetadata, Relation rel, TupleDesc tupleDes
 
 
 /*
- * EndStripeRead finishes a stripe read.
+ * EndStripeRead finishes current stripe read.
+ */
+static void
+EndStripeRead(ColumnarReadState *readState)
```
Comment (jeff-davis) [inline src/backend/columnar/columnar_reader.c:307]: The destructor should just take a pointer to the thing being destructed, the stripeReadState. The memory context can be reset when advancing.

File: src/backend/columnar/columnar_reader.c [hunk 4]
```diff
@@ -204,13 +205,8 @@ ColumnarReadNextRow(ColumnarReadState *readState, Datum *columnValues, bool *col
...
 		{
-			readState->chunkGroupsFiltered +=
-				readState->stripeReadState->chunkGroupsFiltered;
-			readState->currentStripe++;
-			EndStripeRead(readState->stripeReadState);
-			readState->stripeReadState = NULL;
-			MemoryContextReset(readState->stripeReadContext);
-
+			AdvanceStripeRead(readState);
+			EndStripeRead(readState);
```
Comment (jeff-davis) [inline src/backend/columnar/columnar_reader.c:209]: I suggest swapping the order of these.

**Comment by onurctirtir** [comment]
Should we merge this @jeff-davis ?

**Review by jeff-davis — APPROVED** [review]

**Pull request merged**